src/pages/workspace/WorkspaceInitialPage.js

```diff
@@ -127,6 +132,10 @@ const WorkspaceInitialPage = (props) => {
         },
     ];
 
+    if (props.isLoadingReportData && _.isEmpty(props.policy)) {
+        return <FullscreenLoadingIndicator />;
+    }
+
```

Comment:
Sorry for the back and forth @dukenv0307. 

> @mollfpr So we should move the above logic into withPolicy with workspace pages same as withReportOrNotFound?

Seems you understand what I mean yesterday, I don't know what the other things I have are.

I just try create a new HOC for accomodate the return of fullscreen loading if the policy is not provided.

```js
export default function (WrappedComponent) {
    const propTypes = {
        isLoadingReportData: PropTypes.bool,
        ...policyPropTypes,
    };

    const defaultProps = {
        isLoadingReportData: true,
        ...policyDefaultProps,
    };

    const WithPolicyAndFullscreenLoading = (props) => {
        if (props.isLoadingReportData && _.isEmpty(props.policy)) {
            return <FullscreenLoadingIndicator />;
        }

        const rest = _.omit(props, ['forwardedRef']);
        return (
            <WrappedComponent
                // eslint-disable-next-line react/jsx-props-no-spreading
                {...rest}
                ref={props.forwardedRef}
            />
        );
    };

    WithPolicyAndFullscreenLoading.propTypes = propTypes;
    WithPolicyAndFullscreenLoading.defaultProps = defaultProps;
    WithPolicyAndFullscreenLoading.displayName = `WithPolicyAndFullscreenLoading(${getComponentDisplayName(WrappedComponent)})`;

    const withPolicyAndFullscreenLoading = React.forwardRef((props, ref) => (
        <WithPolicyAndFullscreenLoading
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
            forwardedRef={ref}
        />
    ));

    return compose(
        withPolicy,
        withOnyx({
            isLoadingReportData: {
                key: ONYXKEYS.IS_LOADING_REPORT_DATA,
            },
        }),
    )(withPolicyAndFullscreenLoading);
}
```

So we just need to replace `withPolicy` with `withPolicyAndFullscreenLoading` in the workspace pages.

Reply:
@mollfpr Cool, I update right now.